### PR TITLE
[WT-278] feature/right-click-gesture

### PR DESCRIPTION
**Pull request by CandelR**
Right-click gesture that allows a user to:
- Open file previews.
- Generating and deleting shares.
- Uploading, downloading, moving, and deleting items from/to drive and trash.
- Emptying the trash.
- Deleting items permanently.

### CODE REVIEW

File: src/app/backups/components/BackupList/BackupList.tsx [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
+
+  const [selectedBackups, setSelectedBackups] = useState<DeviceBackup[]>([]);
+
+  const onDownloadBackupClicked = async () => {
+    selectedBackups.forEach(async (backup) => dispatch(backupsThunks.downloadBackupThunk(backup)));
   };
-  const onDeleteBackupClicked = async (backup: DeviceBackup) => {
-    dispatch(backupsThunks.deleteBackupThunk(backup));
+  const onDeleteBackupClicked = async () => {
+    selectedBackups.forEach(async (backup) => dispatch(backupsThunks.deleteBackupThunk(backup)));
```
Comment (sg-gs) [inline src/app/backups/components/BackupList/BackupList.tsx]: `async` usage is not required inside forEach as you are not using `await`

File: src/app/drive/components/DriveExplorer/DriveExplorerItem/DriveExplorerListItem/DriveExplorerListItem.tsx [hunk 2]
```diff
@@ -94,37 +96,36 @@ const DriveExplorerListItem = ({ item }: DriveExplorerItemProps): JSX.Element =>
...
     );
   };
   const itemIsShared = item.shares?.length || 0 > 0;
 
   const template = (
-
     <div
       className={`${selectedClassNames} ${isDraggingOverClassNames} ${isDraggingClassNames} file-list-item group`}
-      onContextMenu={onItemRightClicked}
+      // onContextMenu={onItemRightClicked}
```
Comment (sg-gs) [inline src/app/drive/components/DriveExplorer/DriveExplorerItem/DriveExplorerListItem/DriveExplorerListItem.tsx]: Remove this comment

File: src/app/drive/components/DriveExplorer/DriveExplorerItem/DriveExplorerListItem/DriveExplorerListItem.tsx [hunk 3]
```diff
@@ -94,37 +96,36 @@ const DriveExplorerListItem = ({ item }: DriveExplorerItemProps): JSX.Element =>
...
       onClick={onItemClicked}
       onDoubleClick={!item.deleted || !item.isFolder ? onItemDoubleClicked : undefined}
       data-test={`file-list-${item.isFolder ? 'folder' : 'file'}`}
     >
-
       {/* SELECTION */}
-      <div className="box-content flex w-0.5/12 items-center justify-start pl-3">
+      {/* <div className="box-content flex w-0.5/12 items-center justify-start pl-3">
```
Comment (sg-gs) [inline src/app/drive/components/DriveExplorer/DriveExplorerItem/DriveExplorerListItem/DriveExplorerListItem.tsx]: Remove this comment

File: src/app/drive/components/DriveExplorer/DriveExplorerItem/DriveExplorerListItem/DriveExplorerListItem.tsx [hunk 4]
```diff
@@ -187,7 +189,7 @@ const DriveExplorerListItem = ({ item }: DriveExplorerItemProps): JSX.Element =>
       </div>
 
       {/* ACTIONS BUTTON */}
-      <div className="flex w-1/12 items-center">
+      {/* <div className="flex w-1/12 items-center">
```
Comment (sg-gs) [inline src/app/drive/components/DriveExplorer/DriveExplorerItem/DriveExplorerListItem/DriveExplorerListItem.tsx]: Remove this comment

File: src/app/drive/components/EditItemNameDialog/EditItemNameDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        .unwrap()
+        .then(() => {
+          setIsLoading(false);
+          handleOnClose();
+        })
+        .catch((e) => {
+          const errorMessage = e?.message?.includes('already exists') && translate('error.creatingFolder');
+          setError(errorMessage);
+          setIsLoading(false);
+          return e;
```
Comment (sg-gs) [inline src/app/drive/components/EditItemNameDialog/EditItemNameDialog.tsx]: This return is not required as you are not using the error

**Review by sg-gs — CHANGES_REQUESTED** [review]

**Review by sg-gs — APPROVED** [review]

**Review by aboredvaro — CHANGES_REQUESTED** [review]
Issues found:
1. Inconsistent side, top and bottom spacing between lists, follow the [design spacing](https://www.figma.com/file/62hpKnJYlxUN77HUECXAlF/Drive-Web?node-id=3390%3A34184&t=XqLHODSQfWvvyDfI-11).
2. Keyboard shortcuts should work without the right-click menu open too.
3. (Bug) When a **folder** is selected and user taps “enter” on the keyboard, the preview dialog is opened with the name of the folder. When a user taps “enter” on a selected **folder** it should open the folder itself, not the preview.
4. When a user has multiple items selected and opens the right-click menu, "{X} items selected" is not an actionable item, it should be a <div/> like the separators.
5. Some icons in the right-click menu are not the same as in Figma (the download icon, for example), please check them [here](https://www.figma.com/file/62hpKnJYlxUN77HUECXAlF/Drive-Web?node-id=3465%3A72158&t=XqLHODSQfWvvyDfI-11).
6. The three-dot menu is hidden when user opens it on the last item of the list, it should behave similar to right-click menu when it is close to the edges of the screen (top and bottom).
7. When a folder is selected, the option “Open preview” should not appear (only on files), this happens both on three-dot and right-click menus.
8. (Bug) When the user clicks the three-dot icon of an unselected item, select the item and open the three-dot menu. [Check the behavior here](https://inxt.atlassian.net/wiki/spaces/DESIGN/pages/72941572/Right+click#Behavior).
9. On backups section, the empty state is inside the loading list. We need to separate two things: loading list (skeleton) and [empty state](https://www.figma.com/file/62hpKnJYlxUN77HUECXAlF/Drive-Web?node-id=3465%3A42434&t=XqLHODSQfWvvyDfI-11).
10. Drive's list "icon inline actions" (Download, Get link and Move to trash) should be removed since they are replaced by the right-click menu.
11. When user clicks "Rename", it should not use the inline input in the list, it should use the rename dialog.
12. (Bug) When user presses "⬆+F" to open the "New folder" dialog, it writes an "F" at the end of the dialog input. Same happens with the rename dialog when pressing "R".

(Sent a detailed video to Ramón via DM in Slack for better understanding)

**Review by aboredvaro — APPROVED** [review]

**Review by sg-gs — APPROVED** [review]

**Pull request merged**